### PR TITLE
feat: implement events

### LINKED ISSUE #7: Work with events
- repository
- data class
- mock the data from backend(we don't have backend)

### PR DESCRIPTION
**Pull request by Makentoshe**
- implement Category for events

- implement Distance class and EventsRepository

- implement Event, Location and User entities

Closes #7

### CODE REVIEW

File: app/src/main/java/eu/deadbread/eventures/entities/Category.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package eu.deadbread.eventures.entities
+
+sealed class Category {
```
Comment (Indorian1996) [inline app/src/main/java/eu/deadbread/eventures/entities/Category.kt:3]: I suggest to use Category as Data class with name: String and children: List<Category>. Also we can manage color from FB by "colorHex" field. And we can create CategoryType as enum class for cases when we would like change something on our side

File: app/src/main/java/eu/deadbread/eventures/viewmodel/MapScreenViewModel.kt [hunk 2]
```diff
@@ -5,25 +5,45 @@ import androidx.compose.runtime.getValue
...
 
 class MapScreenViewModel(
     private val fusedLocationProviderClient: FusedLocationProviderClient,
+    private val eventsRepository: EventsRepository
 ) : ViewModel() {
 
+    init {
+        viewModelScope.launch {
+            delay(10000)
+            eventsInArea = eventsRepository.getEventsInArea(deviceLocation, Distance.Meters(5000))
```
Comment (Indorian1996) [inline app/src/main/java/eu/deadbread/eventures/viewmodel/MapScreenViewModel.kt:26]: Suggest to add consts file into "common" module and add METERS_IN_KILOMETERS = 1000 and use `5*METERS_IN_KILOMETERS`

**Review by Indorian1996 — CHANGES_REQUESTED** [review]
check comments

File: app/src/main/java/eu/deadbread/eventures/entities/Category.kt (hunk 1 above)
Comment (Makentoshe) [reply app/src/main/java/eu/deadbread/eventures/entities/Category.kt]: I decide to postpone CategoryType until we will be sure we need it

File: app/src/main/java/eu/deadbread/eventures/entities/Category.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package eu.deadbread.eventures.entities
+
+@JvmInline
+value class Color(val colorHex: String)
+data class Category(val title: String, val color: Color, val childs: List<Category> = emptyList())
```
Comment (Indorian1996) [inline app/src/main/java/eu/deadbread/eventures/entities/Category.kt:5]: children
colorHex: String

Comment (Makentoshe) [reply app/src/main/java/eu/deadbread/eventures/entities/Category.kt]: Done

**Review by Indorian1996 — APPROVED** [review]

**Pull request merged**